### PR TITLE
DDF-2899 Update setAttribute method comments in the metacard interface to reflect the expected contract

### PR DESCRIPTION
**Pull request by mcalcote**
#### What does this PR do?
The metacard interface setAttribute method comment only currently mentions setting an Attribute, not using the method to remove an attribute which is the functionality in our primary metacard implementation. The comment should be updated to include this expectation.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@pklinef @stustison @lessarderic @clockard 
#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Core APIs](https://github.com/orgs/codice/teams/core-apis) @ahoffer 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl 
@lessarderic 

#### How should this be tested? (List steps with links to updated documentation)
Inspect the text

#### What are the relevant tickets?
[DDF-2899](https://codice.atlassian.net/browse/DDF-2899)

#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by mcalcote** [comment]
@ricklarsen Can you think of any doc changes needed for this change?

File: catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/Metacard.java [hunk 1]
```diff
@@ -247,7 +247,8 @@
     /**
      * Sets {@link Attribute} with new attribute.
      *
-     * @param attribute new {@link Attribute} to set
+     * @param attribute new {@link Attribute} to set. If <code>null</code> is passed as the Attribute
```
Comment (clockard) [inline catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/Metacard.java]: Should this read something more like `If the getValue() call on the passed in attribute returns <code>null</code>, the attribute will be removed from this {@link Metacard}.`

Comment (coyotesqrl) [inline catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/Metacard.java]: Concur with @clockard. A null `attribute` can't be processed at all.

Comment (mcalcote) [inline catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/Metacard.java]: Oops! Changed.

File: catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/Metacard.java [hunk 2]
```diff
@@ -247,7 +247,8 @@
     /**
      * Sets {@link Attribute} with new attribute.
      *
-     * @param attribute new {@link Attribute} to set
+     * @param attribute new {@link Attribute} to set. If the getValue() call on the passed in
+     * attribute returns <code>null</code>, the attribute will be removed from this {@link Metacard}.
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/Metacard.java]: FYI: Usually in Javadoc, code is marked as `{@code null}`. Both forms generate the same HTML.

**Pull request merged**